### PR TITLE
Check default env against given env config

### PR DESCRIPTION
**Pull request by imjakechapman**
checkenv takes 2 params
1. env path to check default .env against for missing and unset params
2. path to a .env file to override the default root level .env file

### CODE REVIEW

File: src/checkenv.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+export const checkenv = (envPath: string, secondaryEnvPath?: string): IMissingKeys => {
+  const { parsed: envConfig } = dotenv.config({ path: envPath });
+  const { parsed: secondaryEnvConfig } = secondaryEnvPath ? dotenv.config({ path: secondaryEnvPath }) : dotenv.config();
+
+  // Throw errors if we can't find appropriate configs to match
+  if(!envConfig) { throw new Error("Could not find env file with given path to use.") }
+  if(!secondaryEnvConfig) { throw new Error("Could not find the secondary or bases env config.") }
+  
+  const missingKeys = findMissingKeys(envConfig, secondaryEnvConfig);
```
Comment (dmitryvmin) [inline src/checkenv.ts]: The names are little confusing - "secondaryEnvConfig".. maybe baseConfig and featureConfig or something?

File: .env [hunk 2]
```diff
@@ -0,0 +1,3 @@
+SAMPLE_ENV_PARAM=123
+SAMPLE_SET_PARAM=
+EXTRA_PARAM_WE_DONT_NEED=http://seriouslydontneedthisparam.com
```
Comment (dmitryvmin) [inline .env]: yeet

Comment (dmitryvmin) [reply .env]: images work, coool!

File: src/checkenv.ts (hunk 1 above)
Comment (imjakechapman) [reply src/checkenv.ts]: Yeah trying to think of names that describe that the first param is the config file that will be checked against.

`checkedAgainstEnv` and `toCheckEnv` are the most accurate I can come up with.

`toCheckEnv` is the root level `.env` by default and will be able to be overwritten which allow us to check any of the env files inside of the configs directory against the `toCheckEnv` file provided.


Which will mean, for example
`checkedAgainstEnv` - 5 env params
`toCheckEnv` - 3 params, 1 of which are unset

will output

_Discrepancies in (.env | ./configs/uat-instance.env)_
**1** _missing_ param: SOME_PARAM
**2** _unset_ params: SOME_UNSET_PARAM

**Pull request merged**